.github/workflows/checks.yml

```diff
@@ -83,8 +83,7 @@ jobs:
 
       - name: "Unit tests"
         run: |
-          go test ./...
-
+          go test -timeout 20m ./...
```

Comment:
This more than doubles our "Quick Check" time:

https://github.com/opentofu/opentofu/actions/runs/9506261377
vs
https://github.com/opentofu/opentofu/actions/runs/9504270127

It adds 1 minute of overhead to Unit Tests linux_amd64, but adds 10!! minutes to linux_arm.  If we adopt this, we can not run these checks on emulated architectures.